src/Mvc/Mvc.Core/src/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -103,23 +99,9 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
             }
 
             var value = result.Value;
-
-            if (value != null && _asyncEnumerableReaderFactory.TryGetReader(value.GetType(), out var reader))
-            {
-                return ExecuteAsyncEnumerable(context, result, value, reader);
-            }
-
             return ExecuteAsyncCore(context, result, objectType, value);
         }
 
-        private async Task ExecuteAsyncEnumerable(ActionContext context, ObjectResult result, object asyncEnumerable, Func<object, Task<ICollection>> reader)
-        {
-            Log.BufferingAsyncEnumerable(Logger, asyncEnumerable);
-
-            var enumerated = await reader(asyncEnumerable);
-            await ExecuteAsyncCore(context, result, enumerated.GetType(), enumerated);
-        }
-
```

Comment:
By doing this, are we not removing support for this on other formatters? (JSON.NET, XML)

Reply:
That was one of the options I considered. The current implementation is also a bit of a whack-a-mole in that it only works for all the result types that we account for. Custom result types would have the same issue (e.g. if a user makes a XmlResult or a ProtobufResult). Given the choice of either handling it at per-formatter vs per-result, the latter seems like a better choice. 

I can make an announcement since this is a breaking behavior change.

Reply:
Result types derived from ObjectResult would have gotten this for free right?

Reply:
In the ordinary case, yes. It's possible that there's also custom result executors that would negate it though.